### PR TITLE
Remove DisableWindowsRenderThread and always use the threaded renderer

### PR DESCRIPTION
**Pull request by abcdefg30**
Works around the window maximisation/restoring issue on windows by not ticking `RenderTick` while the window is suspended. This should additionally improve performance when the window is suspended. (Not to mention the performance improvement on Windows by using the threaded renderer!)

It turns out our `RenderTick` somehow consumed SDL events before `PumpInput` could restore the window. Skipping the render and explicitly calling `PumpInput` works around that issue.

### CODE REVIEW

File: OpenRA.Game/Game.cs
Comment (sorcerer86pt) [inline OpenRA.Game/Game.cs]: can you add link to this bug for documentation reference?

**Review by sorcerer86pt — DISMISSED** [review]

File: OpenRA.Game/Game.cs
Comment (abcdefg30) [reply OpenRA.Game/Game.cs]: Good idea actually.

**Comment by abcdefg30** [comment]
I want to have another look here and change some things, so marking this as "on hold".

File: OpenRA.Platforms.Default/Sdl2Input.cs
Comment (pchote) [inline OpenRA.Platforms.Default/Sdl2Input.cs]: IMO the naming duplication here isn't great, or having the input logic reach into and change state inside Game.

Can we please have this flag live inside the platform window, and be exposed via a readonly `IPlatformWindow.Minimized` property? The Game loop can then call `Renderer.WindowMinimized` which returns `Window.Minimized` the same way it proxies the other window-specific things.

Comment (abcdefg30) [reply OpenRA.Platforms.Default/Sdl2Input.cs]: This is exactly how `Game.HasInputFocus` works just a few lines above. But I'll look into your request.

Comment (pchote) [reply OpenRA.Platforms.Default/Sdl2Input.cs]: This would be a good opportunity to fix that too :P

Comment (abcdefg30) [reply OpenRA.Platforms.Default/Sdl2Input.cs]: Did that in #17909. This here can then use the same approach.

**Comment by pchote** [comment]
What do you want to do here @abcdefg30? Should we close and park on the Future milestone?

**Comment by abcdefg30** [comment]
I wanted to look at an issue with maximizing again and need to adapt this to #17909. Thinking about it, I should probably split the "don't render when minimized/suspended" part off into a new PR. (#18354)

**Comment by pchote** [comment]
No progress here in a real long time, so moving to the Future milestone as mentioned above.

**Comment by abcdefg30** [comment]
Updated after #18670 was merged.

**Comment by pchote** [comment]
Unfortunately I can still reproduce the original bug on this PR in my VM 😞 

Restoring from minimized seems to work reliably, but clicking on the taskbar icon to minimize the visible window often gives the alert tone instead of minimizing.

**Comment by abcdefg30** [comment]
Yep, this wasn't fixed in the initial version of this PR either and won't until we/I figure out what in our `RenderTick` method is consuming the events. I figured that minimizing the window via the task bar is not critical since you can always do that via the button in the top window bar, and not enough of a use-case to justify holding the improvements from the threaded rendering up. If you disagree we'll have to put this on hold/close it until the root of the issue is fixed.

**Comment by pchote** [comment]
I'd like to first have a chance to experiment with glfw to see whether we can sidestep the event and the cursor problems completely. If that doesn't work (or nobody actually tries it within a reasonable time) then IMO the best compromise here would be to enable the threaded renderer for the fullscreen mode but for now keep the fallback for windowed. This will give the threaded benefits for the majority of players without risking any workflow regressions for the small number who do use windowed.

**Comment by pchote** [comment]
I did a rough prototype with GLFW and found that suffers the same problems. So can we take the safer compromise of using threaded renderer for (pseudo-)fullscreen and non-threaded for windowed?

**Comment by pchote** [comment]
Ping @abcdefg30. Are you okay with implementing my windowed vs (pseudo)fullscreen suggestion?

**Comment by abcdefg30** [comment]
Updated so that the threaded rendering is still disabled for windowed mode on Windows.

**Review by pchote — DISMISSED** [review]

**Review by penev92 — DISMISSED** [review]
I can confirm a noticeable increase in FPS without regressing the minimizing.
I would remove the "from the taskbar" part of the comment because it also affects the WindowsButton+DownArrow key combo.

**Comment by abcdefg30** [comment]
Updated the comment.

**Review by penev92 — APPROVED** [review]
👍

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6cb348fae63c2f05b6846428036f8cff8745c668

**Pull request merged**
